libs/react-helpers/src/hooks/use-data-provider.ts

```diff
@@ -32,30 +48,48 @@ export function useDataProvider<Data, Delta>(
       reloadRef.current(force);
     }
   }, []);
+  const load = useCallback((pagination: Pagination) => {
+    if (loadRef.current) {
+      return loadRef.current(pagination);
+    }
+    return Promise.reject();
+  }, []);
   const callback = useCallback(
-    ({ data, error, loading, delta }) => {
+    ({ data, error, loading, delta, insertData, totalCount }) => {
       setError(error);
       setLoading(loading);
       if (!error && !loading) {
-        // if update function returns true it means that component handles updates
+        // if update or insert function returns true it means that component handles updates
         // component can use flush() which will call callback without delta and cause data state update
-        if (!initialized.current || !delta || !update || !update(delta)) {
-          initialized.current = true;
-          setData(data);
+        if (initialized.current) {
+          if (delta && update && update({ delta, data })) {
```

Comment:
nit (non-blocking): one thing this which makes this harder-than-it-has-to-be to digest is that the update and the insert return a boolean. 
- if they send the data to somewhere, they probably shouldn't be called as a condition of an if statement, but in the body of it
- if they don't send the data anywhere, tit would be less confusing if  they would be called `shouldUpdate` and `shouldInsert`, or something similar to indicate their function a bit more clearly

Reply:
i can't say i understood the points regarding ag-grid, my comment was purely related to naming and separation of concerns :)